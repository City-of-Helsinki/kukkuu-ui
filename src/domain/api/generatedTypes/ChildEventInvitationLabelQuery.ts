/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ChildEventInvitationLabelQuery
// ====================================================

export interface ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node_EventGroupNode {
  /**
   * The ID of the object.
   */
  id: string;
  canChildEnroll: boolean | null;
}

export interface ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node_EventNode {
  /**
   * The ID of the object.
   */
  id: string;
  canChildEnroll: boolean | null;
}

export type ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node =

    | ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node_EventGroupNode
    | ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node_EventNode;

export interface ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges {
  /**
   * The item at the end of the edge
   */
  node: ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node | null;
}

export interface ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups {
  /**
   * Contains the nodes in this connection.
   */
  edges: (ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges | null)[];
}

export interface ChildEventInvitationLabelQuery_child {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * All upcoming events and event groups for the child's project.
   */
  upcomingEventsAndEventGroups: ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups | null;
}

export interface ChildEventInvitationLabelQuery {
  child: ChildEventInvitationLabelQuery_child | null;
}

export interface ChildEventInvitationLabelQueryVariables {
  childId: string;
}
