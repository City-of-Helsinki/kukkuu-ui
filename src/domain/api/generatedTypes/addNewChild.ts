/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AddChildMutationInput } from './globalTypes';

// ====================================================
// GraphQL mutation operation: addNewChild
// ====================================================

export interface addNewChild_addChild_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  year: number;
}

export interface addNewChild_addChild_child {
  /**
   * The ID of the object.
   */
  id: string;
  firstName: string;
  lastName: string;
  birthdate: any;
  postalCode: string;
  project: addNewChild_addChild_child_project;
}

export interface addNewChild_addChild {
  child: addNewChild_addChild_child | null;
}

export interface addNewChild {
  /**
   * This mutation cannot be used before one has started using the service with "SubmitChildrenAndGuardianMutation".
   */
  addChild: addNewChild_addChild | null;
}

export interface addNewChildVariables {
  input: AddChildMutationInput;
}
