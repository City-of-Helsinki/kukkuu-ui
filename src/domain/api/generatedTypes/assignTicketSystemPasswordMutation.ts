/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssignTicketSystemPasswordMutationInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: assignTicketSystemPasswordMutation
// ====================================================

export interface assignTicketSystemPasswordMutation_assignTicketSystemPassword {
  /**
   * The assigned ticket system password
   */
  password: string | null;
}

export interface assignTicketSystemPasswordMutation {
  assignTicketSystemPassword: assignTicketSystemPasswordMutation_assignTicketSystemPassword | null;
}

export interface assignTicketSystemPasswordMutationVariables {
  input: AssignTicketSystemPasswordMutationInput;
}
