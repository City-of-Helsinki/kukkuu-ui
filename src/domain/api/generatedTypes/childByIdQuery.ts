/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { EventParticipantsPerInvite, RelationshipTypeEnum } from "./globalTypes";

// ====================================================
// GraphQL query operation: childByIdQuery
// ====================================================

export interface childByIdQuery_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  year: number;
}

export interface childByIdQuery_child_occurrences_edges_node_venue {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  address: string | null;
}

export interface childByIdQuery_child_occurrences_edges_node_event {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  /**
   * In minutes
   */
  duration: number | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
}

export interface childByIdQuery_child_occurrences_edges_node_enrolments_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * An unique encoded reference id
   */
  referenceId: string | null;
}

export interface childByIdQuery_child_occurrences_edges_node_enrolments_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_occurrences_edges_node_enrolments_edges_node | null;
}

export interface childByIdQuery_child_occurrences_edges_node_enrolments {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_occurrences_edges_node_enrolments_edges | null)[];
}

export interface childByIdQuery_child_occurrences_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  time: any;
  venue: childByIdQuery_child_occurrences_edges_node_venue;
  event: childByIdQuery_child_occurrences_edges_node_event;
  enrolments: childByIdQuery_child_occurrences_edges_node_enrolments;
}

export interface childByIdQuery_child_occurrences_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_occurrences_edges_node | null;
}

export interface childByIdQuery_child_occurrences {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_occurrences_edges | null)[];
}

export interface childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventNode {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
}

export interface childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventGroupNode {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
}

export type childByIdQuery_child_availableEventsAndEventGroups_edges_node = childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventNode | childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventGroupNode;

export interface childByIdQuery_child_availableEventsAndEventGroups_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_availableEventsAndEventGroups_edges_node | null;
}

export interface childByIdQuery_child_availableEventsAndEventGroups {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_availableEventsAndEventGroups_edges | null)[];
}

export interface childByIdQuery_child_pastEvents_edges_node_occurrences_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  time: any;
}

export interface childByIdQuery_child_pastEvents_edges_node_occurrences_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_pastEvents_edges_node_occurrences_edges_node | null;
}

export interface childByIdQuery_child_pastEvents_edges_node_occurrences {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_pastEvents_edges_node_occurrences_edges | null)[];
}

export interface childByIdQuery_child_pastEvents_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
  occurrences: childByIdQuery_child_pastEvents_edges_node_occurrences;
}

export interface childByIdQuery_child_pastEvents_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_pastEvents_edges_node | null;
}

export interface childByIdQuery_child_pastEvents {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_pastEvents_edges | null)[];
}

export interface childByIdQuery_child_relationships_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  type: RelationshipTypeEnum | null;
}

export interface childByIdQuery_child_relationships_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_relationships_edges_node | null;
}

export interface childByIdQuery_child_relationships {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_relationships_edges | null)[];
}

export interface childByIdQuery_child {
  /**
   * The ID of the object.
   */
  id: string;
  firstName: string;
  lastName: string;
  birthdate: any;
  postalCode: string;
  project: childByIdQuery_child_project;
  occurrences: childByIdQuery_child_occurrences;
  availableEventsAndEventGroups: childByIdQuery_child_availableEventsAndEventGroups | null;
  pastEvents: childByIdQuery_child_pastEvents | null;
  relationships: childByIdQuery_child_relationships;
}

export interface childByIdQuery {
  /**
   * The ID of the object
   */
  child: childByIdQuery_child | null;
}

export interface childByIdQueryVariables {
  id: string;
}
