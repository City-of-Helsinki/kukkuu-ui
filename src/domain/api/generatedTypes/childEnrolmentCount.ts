/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ChildEnrolmentCount
// ====================================================

export interface ChildEnrolmentCount_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * How many times a single user can participate events per year. Changing this will not affect any existing enrolments.
   */
  enrolmentLimit: number;
}

export interface ChildEnrolmentCount_child {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * How many past enrolments child has this year.
   */
  pastEnrolmentCount: number | null;
  project: ChildEnrolmentCount_child_project;
}

export interface ChildEnrolmentCount {
  child: ChildEnrolmentCount_child | null;
}

export interface ChildEnrolmentCountVariables {
  childId: string;
}
