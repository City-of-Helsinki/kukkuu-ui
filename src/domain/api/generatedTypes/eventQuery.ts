/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { EventParticipantsPerInvite } from "./globalTypes";

// ====================================================
// GraphQL query operation: eventQuery
// ====================================================

export interface eventQuery_event_occurrences_edges_node_event {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
}

export interface eventQuery_event_occurrences_edges_node_venue {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  address: string | null;
}

export interface eventQuery_event_occurrences_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  time: any;
  remainingCapacity: number | null;
  event: eventQuery_event_occurrences_edges_node_event;
  venue: eventQuery_event_occurrences_edges_node_venue;
  childHasFreeSpotNotificationSubscription: boolean | null;
}

export interface eventQuery_event_occurrences_edges {
  /**
   * The item at the end of the edge
   */
  node: eventQuery_event_occurrences_edges_node | null;
}

export interface eventQuery_event_occurrences {
  /**
   * Contains the nodes in this connection.
   */
  edges: (eventQuery_event_occurrences_edges | null)[];
}

export interface eventQuery_event {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  description: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
  /**
   * In minutes
   */
  duration: number | null;
  capacityPerOccurrence: number;
  occurrences: eventQuery_event_occurrences;
}

export interface eventQuery {
  /**
   * The ID of the object
   */
  event: eventQuery_event | null;
}

export interface eventQueryVariables {
  id: string;
  date?: any | null;
  time?: any | null;
  childId?: string | null;
}
