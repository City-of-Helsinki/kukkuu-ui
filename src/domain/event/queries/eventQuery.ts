import gql from 'graphql-tag';

// TODO: Add remainingCapacity to occurrences when backend bug is fixed
const eventQuery = gql`
  query eventQuery($id: ID!) {
    event(id: $id) {
      id
      name
      description
      shortDescription
      image
      participantsPerInvite
      duration
      capacityPerOccurrence
      occurrences {
        edges {
          node {
            id
            time
            venue {
              name
              description
              address
            }
          }
        }
      }
    }
  }
`;
export default eventQuery;
