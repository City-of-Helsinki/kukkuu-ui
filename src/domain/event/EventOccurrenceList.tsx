import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import { eventQuery_event_occurrences as Occurrences } from '../api/generatedTypes/eventQuery';
import EventOccurrence from './EventOccurrence';
import styles from './eventOccurrenceList.module.scss';

interface EventOccurrenceListProps {
  occurrences: Occurrences;
  eventId: string;
}

const EventOccurrenceList: FunctionComponent<EventOccurrenceListProps> = ({
  occurrences,
}) => {
  const { t } = useTranslation();
  return (
    <table className={styles.eventOccurrenceList}>
      <tbody>
        <tr className={styles.mobileHeader}>
          <th>{t('event.register.occurrenceTableHeader.eventInformation')}</th>
          <th>{t('event.register.occurrenceTableHeader.freePlaces')}</th>
        </tr>

        <tr className={styles.desktopHeader}>
          <th>{t('event.register.occurrenceTableHeader.date')}</th>
          <th>{t('event.register.occurrenceTableHeader.time')}</th>
          <th>{t('event.register.occurrenceTableHeader.venue')}</th>
          <th>{t('event.register.occurrenceTableHeader.freePlaces')}</th>
          <th></th>
        </tr>

        {occurrences.edges.map(edge =>
          edge?.node ? (
            <EventOccurrence key={edge.node.id} occurrence={edge.node} />
          ) : null
        )}
      </tbody>
    </table>
  );
};

export default EventOccurrenceList;
