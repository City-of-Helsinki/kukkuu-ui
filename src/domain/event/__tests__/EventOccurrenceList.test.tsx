import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { eventQuery_event_occurrences as Occurrences } from '../../api/generatedTypes/eventQuery';
import EventOccurrenceList from '../EventOccurrenceList';

const mockedOccurrences: Occurrences = {
  edges: [
    {
      node: {
        id: 'T2NjdXJyZW5jZU5vZGU6Mg==',
        time: '2020-03-08T04:00:00+00:00',
        remainingCapacity: 99,
        event: {
          id: 'asdf',
          name: 'asdf',
        },
        venue: {
          id: 'fdsa',
          name: 'Musiikkitalo',
          description: 'Helsingin oma Musiikkitalo',
          address: 'Mannerheimintie 13A, Helsinki',
        },
      },
    },
    {
      node: {
        id: 'T2NjdXJyZW5jZU5vZGU6Mg==',
        time: '2020-04-08T04:00:00+00:00',
        remainingCapacity: 88,
        event: {
          id: 'asdf',
          name: 'asdf',
        },
        venue: {
          id: 'fdsa1',
          name: 'Somewhere',
          description: 'Under the',
          address: 'Rainbow 123',
        },
      },
    },
  ],
};

it('renders snapshot correctly', () => {
  const element = shallow(
    <EventOccurrenceList
      occurrences={mockedOccurrences}
      eventId={'sksksksksksk'}
    />
  );
  expect(toJson(element)).toMatchSnapshot();
});
