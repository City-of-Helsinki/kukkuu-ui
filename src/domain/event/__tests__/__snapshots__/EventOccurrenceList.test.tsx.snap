// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<table
  className="eventOccurrenceList"
>
  <tbody>
    <tr
      className="mobileHeader"
    >
      <th>
        Tapahtuman tiedot
      </th>
      <th>
        Vapaat paikat
      </th>
    </tr>
    <tr
      className="desktopHeader"
    >
      <th>
        Päivämäärä
      </th>
      <th>
        Kellonaika
      </th>
      <th>
        Tapahtumapaikka
      </th>
      <th>
        Vapaat paikat
      </th>
      <th />
    </tr>
    <EventOccurrence
      key="T2NjdXJyZW5jZU5vZGU6Mg=="
      occurrence={
        Object {
          "event": Object {
            "id": "asdf",
            "name": "asdf",
          },
          "id": "T2NjdXJyZW5jZU5vZGU6Mg==",
          "remainingCapacity": 99,
          "time": "2020-03-08T04:00:00+00:00",
          "venue": Object {
            "address": "Mannerheimintie 13A, Helsinki",
            "description": "Helsingin oma Musiikkitalo",
            "id": "fdsa",
            "name": "Musiikkitalo",
          },
        }
      }
    />
    <EventOccurrence
      key="T2NjdXJyZW5jZU5vZGU6Mg=="
      occurrence={
        Object {
          "event": Object {
            "id": "asdf",
            "name": "asdf",
          },
          "id": "T2NjdXJyZW5jZU5vZGU6Mg==",
          "remainingCapacity": 88,
          "time": "2020-04-08T04:00:00+00:00",
          "venue": Object {
            "address": "Rainbow 123",
            "description": "Under the",
            "id": "fdsa1",
            "name": "Somewhere",
          },
        }
      }
    />
  </tbody>
</table>
`;
